graph/src/data/subgraph/schema.rs

```diff
@@ -1181,11 +1190,35 @@ pub fn attribute_index_definitions(
     indexing_ops
 }
 
+fn is_derived_field(field: &Field) -> bool {
+    field
+        .directives
+        .iter()
+        .any(|dir| dir.name == Name::from("derivedFrom"))
+}
+
+fn is_entity(type_name: &str, definitions: &[Definition]) -> bool {
+    definitions.iter().any(|defn| {
+        if let Definition::TypeDefinition(TypeDefinition::Object(schema_object)) = defn {
```

Comment:
As soon as we add #927, this check will be wrong. The following check is more robust and would be good to adopt here (even if it's copied in):

https://github.com/graphprotocol/graph-node/blob/master/graphql/src/schema/ast.rs#L377-L394

Reply:
Good point. Just fixed that. Longer term, we should think about how to restructure things so we do not have to have this logic in two places, since it's easy right now to make a change in only one and forget the other.